docs/Makefile

```diff
@@ -0,0 +1,7 @@
+docs: Manifest.toml
+	julia --project=. make.jl
```

Comment:
If I'm understanding correctly how to test this, I'm testing it by running 
```
cd docs
make
```
It seems like it works, but throws a bunch of warnings.

Do we also want to support `make docs` from top-level?

Reply:
you should be able to do `make -C docs`

Reply:
Can we add a usage comment at the top of each one? It might be nice to also note from which directory the command should be used

Reply:
oh yeah. That works too and reduces some of the warnings. We should perhaps add it to the `README`.

Do we care about fixing these warnings? These are the ones I see.
From the `docs/` directory:
```
┌ Warning: unknown command \Join
└ @ MathTeXEngine ~/.julia/packages/MathTeXEngine/ZP0gS/src/parser/commands_registration.jl:48
┌ Warning: precompile directive
│     precompile(fbody, (Nothing, Nothing, Nothing, Nothing, Nothing, Al, Tuple{Bool, Bool}, Rect2f, Auto, Auto, Bool, Bool, Symbol, Symbol, Float64, Float64, Emptykwargs, Type{GridLayout}, Int, Int))
│ failed. Please report an issue in GridLayoutBase (after checking for duplicates) or remove this directive.
└ @ GridLayoutBase ~/.julia/packages/GridLayoutBase/CSLqD/src/precompile.jl:31
┌ Warning: precompile directive
│     precompile(fbody, (Nothing, Nothing, Nothing, Nothing, Nothing, Al, Tuple{Bool, Bool}, Rect2f, Auto, Auto, Bool, Bool, Symbol, Symbol, Float64, Float64, Emptykwargs, Type{GridLayout}, Int, Int))
│ failed. Please report an issue in GridLayoutBase (after checking for duplicates) or remove this directive.
└ @ GridLayoutBase ~/.julia/packages/GridLayoutBase/CSLqD/src/precompile.jl:31
┌ Warning: precompile directive
│     precompile(fbody, (Nothing, Nothing, Nothing, Nothing, Nothing, Al, Tuple{Bool, Bool}, Rect2f, Auto, Auto, Bool, Bool, Symbol, Symbol, Float64, Float64, Emptykwargs, Type{GridLayout}, Int, Int))
│ failed. Please report an issue in GridLayoutBase (after checking for duplicates) or remove this directive.
└ @ GridLayoutBase ~/.julia/packages/GridLayoutBase/CSLqD/src/precompile.jl:31
[ Info: generating markdown page from `~/CliMA/ClimaCore.jl/docs/tutorials/introduction.jl`
[ Info: writing result to `~/CliMA/ClimaCore.jl/docs/src/tutorials/introduction.md`
[ Info: generating notebook from `~/CliMA/ClimaCore.jl/docs/tutorials/introduction.jl`
[ Info: writing result to `~/CliMA/ClimaCore.jl/docs/src/tutorials/introduction.ipynb`
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: ExpandTemplates: expanding markdown templates.
[ Info: Expanding bibliography.
[ Info: Expanding bibliography entry: Berrut2004.
[ Info: Expanding bibliography entry: Nair2005.
[ Info: Expanding bibliography entry: Rancic1996.
[ Info: Expanding bibliography entry: Ronchi1996.
[ Info: Expanding bibliography entry: Taylor2010.
[ Info: CrossReferences: building cross-references.
[ Info: Citations: building citations.
[ Info: Expanding citation: Ronchi1996.
[ Info: Expanding citation: Rancic1996.
[ Info: Expanding citation: Nair2005.
[ Info: Expanding citation: Rancic1996.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Taylor2010.
[ Info: Expanding citation: Taylor2010.
[ Info: Expanding citation: Taylor2010.
[ Info: CheckDocument: running document checks.
┌ Warning: 9 docstrings not included in the manual:
│ 
│     ClimaCore.DataLayouts.VIFH
│     ClimaCore.RecursiveApply.tuplemap :: Union{Tuple{F}, Tuple{F, Tuple, Tuple}} where F
│     ClimaCore.RecursiveApply.tuplemap :: Union{Tuple{F}, Tuple{F, Tuple}} where F
│     ClimaCore.DataLayouts.IF
│     ClimaCore.DataLayouts.IFH :: Union{Tuple{Ni}, Tuple{S}, Tuple{Any, Any}} where {S, Ni}
│     ClimaCore.DataLayouts.IFH
│     ClimaCore.DataLayouts.VIJFH
│     ClimaCore.DataLayouts.VF
│     ClimaCore.DataLayouts.IJF
│ 
│ 
│ These are docstrings in the checked modules (configured with the modules keyword)
│ that are not included in @docs or @autodocs blocks.
└ @ Documenter.DocChecks ~/.julia/packages/Documenter/qdbx6/src/DocChecks.jl:70
[ Info: Populate: populating indices.
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
┌ Warning: couldn't find file "/home/valeria/CliMA/ClimaCore.jl/docs/src/tutorials/<unknown>/tutorials/introduction.jl" when generating URL
└ @ Documenter.Utilities ~/.julia/packages/Documenter/qdbx6/src/Utilities/Utilities.jl:424
┌ Warning: Documenter could not auto-detect the building environment Skipping deployment.
└ @ Documenter ~/.julia/packages/Documenter/qdbx6/src/deployconfig.jl:75
```

From top-level:
```
$ make -C docs
make: Entering directory '/home/valeria/CliMA/ClimaCore.jl/docs'
julia --project=. make.jl
[ Info: generating markdown page from `~/CliMA/ClimaCore.jl/docs/tutorials/introduction.jl`
[ Info: writing result to `~/CliMA/ClimaCore.jl/docs/src/tutorials/introduction.md`
[ Info: generating notebook from `~/CliMA/ClimaCore.jl/docs/tutorials/introduction.jl`
[ Info: writing result to `~/CliMA/ClimaCore.jl/docs/src/tutorials/introduction.ipynb`
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: ExpandTemplates: expanding markdown templates.
[ Info: Expanding bibliography.
[ Info: Expanding bibliography entry: Berrut2004.
[ Info: Expanding bibliography entry: Nair2005.
[ Info: Expanding bibliography entry: Rancic1996.
[ Info: Expanding bibliography entry: Ronchi1996.
[ Info: Expanding bibliography entry: Taylor2010.
[ Info: CrossReferences: building cross-references.
[ Info: Citations: building citations.
[ Info: Expanding citation: Ronchi1996.
[ Info: Expanding citation: Rancic1996.
[ Info: Expanding citation: Nair2005.
[ Info: Expanding citation: Rancic1996.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Berrut2004.
[ Info: Expanding citation: Taylor2010.
[ Info: Expanding citation: Taylor2010.
[ Info: Expanding citation: Taylor2010.
[ Info: CheckDocument: running document checks.
┌ Warning: 9 docstrings not included in the manual:
│ 
│     ClimaCore.DataLayouts.IF
│     ClimaCore.DataLayouts.IJF
│     ClimaCore.DataLayouts.VIFH
│     ClimaCore.DataLayouts.VIJFH
│     ClimaCore.DataLayouts.IFH :: Union{Tuple{Ni}, Tuple{S}, Tuple{Any, Any}} where {S, Ni}
│     ClimaCore.DataLayouts.IFH
│     ClimaCore.DataLayouts.VF
│     ClimaCore.RecursiveApply.tuplemap :: Union{Tuple{F}, Tuple{F, Tuple, Tuple}} where F
│     ClimaCore.RecursiveApply.tuplemap :: Union{Tuple{F}, Tuple{F, Tuple}} where F
│ 
│ 
│ These are docstrings in the checked modules (configured with the modules keyword)
│ that are not included in @docs or @autodocs blocks.
└ @ Documenter.DocChecks ~/.julia/packages/Documenter/qdbx6/src/DocChecks.jl:70
[ Info: Populate: populating indices.
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
┌ Warning: couldn't find file "/home/valeria/CliMA/ClimaCore.jl/docs/src/tutorials/<unknown>/tutorials/introduction.jl" when generating URL
└ @ Documenter.Utilities ~/.julia/packages/Documenter/qdbx6/src/Utilities/Utilities.jl:424
┌ Warning: Documenter could not auto-detect the building environment Skipping deployment.
└ @ Documenter ~/.julia/packages/Documenter/qdbx6/src/deployconfig.jl:75
make: Leaving directory '/home/valeria/CliMA/ClimaCore.jl/docs'
```

Reply:
Yes, it would be good to fix warnings: I've been chipping away at them every time i update the docs, but there is still a bit to do.